### PR TITLE
Scene UIDs bug

### PR DESCRIPTION
**Pull request by AlbertVVila**
# The Bug
- When a scene was modified, all the GameObject _UIDs_ where modified, even if we only changed a single value.

# The Solution
- Now we only modify _UIDs_ for Added Scenes on top of the current scene. So, the only changes that will be shown when a scene is modified are the current changes done and nothing else.

# The Test
- Test with the _Forbidden Temple_ scene. Try to open the scene, save it and check with the _Visual Studio_ comparator if any change is shown.
- Then Add a GameObject, save the scene and check in the comparator that the only modification is the gameobejct added.
- Test Adding a scene like the _Hearse_ or the _DollHouse_ and check again with the _Visual Studio_ comparator.

### CODE REVIEW

File: Source/ResourceScene.cpp [hunk 1]
```diff
@@ -163,8 +194,7 @@ bool ResourceScene::Load()
 
 void ResourceScene::AssignNewUUID(GameObject* go, unsigned parentUID)
 {
-
-	if (go != App->scene->root && go != App->scene->canvas)
+	if (go->UUID != 0 && go->UUID != 1)
```
Comment (Rubetoman) [inline Source/ResourceScene.cpp]: I miss u.

File: Source/ResourceScene.cpp [hunk 2]
```diff
@@ -99,7 +99,10 @@ bool ResourceScene::Load()
 	JSON *json = new JSON(data);
 	JSON_value* gameobjectsJSON = json->GetValue("GameObjects");
 	std::map<unsigned, GameObject*> gameobjectsMap; //Necessary to assign parent-child efficiently
-	gameobjectsMap.insert(std::pair<unsigned, GameObject*>(App->scene->canvas->UUID, App->scene->canvas));
+	GameObject* sceneRoot = new GameObject("World", 0);
+	GameObject* sceneCanvas = new GameObject("Canvas", App->scene->canvas->UUID);
+	sceneCanvas->parentUUID = -1;
```
Comment (Rubetoman) [inline Source/ResourceScene.cpp]: A letter is missing here.

File: Source/ResourceScene.cpp (hunk 1 above)
Comment (AlbertVVila) [reply Source/ResourceScene.cpp]: i miss u too 😄  (i'm gonna correct it)

Comment (AlfonsoMestres) [reply Source/ResourceScene.cpp]: 🍆 ❤️

**Pull request merged**